### ISSUE TITLE
Cleanup tool/internal service for resources of crashed applications

### ISSUE THREAD

**Issue by elfenpiff** [eclipse-iceoryx/iceoryx2#21]
## Brief feature description

When an application crashes, it can happen that a service is only partially removed and becomes unusable. 
An internal service and/or a command line application shall cleanup the remainders of such services.

The following use cases shall be considered:

* remove static service info, when only the dynamic service info exists
* remove dynamic service info, when only the static service info exists
* remove the data segment of a publisher if the owning process no longer exists
* remove the publisher/subscriber as participant from the dynamic service info when the owning process no longer exists.

**Comment by elfenpiff** [comment]
Previous discussions:

@elBoberido : 
Could this be done on the startup of the application?

@elfenpiff : 
Yes. The basic idea would be to implement a "function" that does it in a thread-safe manner and then either everyone can do this in a decentral fashion or one central daemon or command line tool does this.

This is then up to the deployment/user.

**Comment by elfenpiff** [comment]
cc @orecham This would be something for our cli tool `iox2`

**Comment by elBoberido** [comment]
@elfenpiff is there currently a way to find out which domains are in use?

**Comment by elfenpiff** [comment]
@elBoberido, nope. I think we will never be able to. The different domains have the goal to completely separate the iceoryx2 instances, if there would be somehow a domain registry, the domains would again be connected. But what can be done is, to provide a specific domain and take a look if there is somthing.

**Comment by elBoberido** [comment]
@elfenpiff I didn't meant to have a domain registry but more something like a pre-defined folder where each domain creates just a file with the domain id. A tool could then list all domains and also check if there are still active services in the domain. Not sure if it is really a good idea since one might not want that someone can figure out that a specific domain is used.

**Comment by elBoberido** [comment]
Another option would be to just recreate the folder structure for the domains in a specific folder to look this up.

For example there would be a `/foo`, `/tmp/bar` and `/tmp/iceoryx2`, then the node would create those folder in e.g. `/tmp/iceoryx2-domains` like
```
/tmp/iceoryx2-domains/foo
/tmp/iceoryx2-domains/tmp/bar
/tmp/iceoryx2-domains/tmp/iceoryx2
```

This would easily allow a tool to find the correct folders. We would then also need to obtain the prefix and shm directory from somewhere but those information could be stored in the static config.
